module.nix

```diff
@@ -193,7 +204,7 @@ in
             UMask = "0027";
             # TODO /run might be sufficient
             WorkingDirectory = "%S/authentik";
-            DynamicUser = true;
+            User = "authentik";
```

Comment:
I assume this was done to allow all services to read the media uploads?
Any particular reason why this wasn't solved over e.g. SupplementaryGroups/ReadWritePaths and group-level permissions?

The nice thing about `DynamicUser` is that it implies a few other sandboxing mechanisms out of the box (e.g. `PrivateTmp` & `RemoveIPC`, the full list of other settings influenced by that can be found in `systemd.exec(5)`), so if this gets solved via `User=`, I'd prefer to re-add those.

Reply:
I've reworked this so that we create the folder within the preStart script and not with tmpfilesd.

Reply:
@xanderio correct me if I'm wrong, but I don't think this line is needed anymore?

Reply:
Totally missed that.

Reply:
We can't make use of `DynamicUser` in this case, as the uid of the `media` directory needs to stay the same. I'm going to readd a static `authentik` user and group, but this time set `PrivateTmp` and `RemoveIPC` manually on the units. This then also allows us to make use of tmpfiled to create the `media` directory.

Reply:
> as the uid of the media directory needs to stay the same

May I ask which issues you encountered? Would be curious to see what the problem is.

Reply:
To quote the documentation of `DynamicUser`: 

> However, UID/GIDs are recycled after a unit is terminated. Care should be taken that any processes running as part of a unit for which dynamic users/groups are enabled do not leave files or directories owned by these users/groups around, as a different unit might get the same UID/GID assigned later on, and thus gain access to these files or directories


This means that UID which owns the directory may change across unit or system restarts. After this authentik will no longer be able to access any of it's media files.

Reply:
> This means that UID which owns the directory may change across unit or system restarts. After this authentik will no longer be able to access any of it's media files.

According to https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#RuntimeDirectory= a recursive chown will be done whenever needed:

> If the specified directories already exist and their owning user or group do not match the configured ones, all files and directories below the specified directories as well as the directories themselves will have their file ownership recursively changed to match what is configured

Otherwise, the combination of `StateDirectory`/`DynamicUser` would never work.

Reply:
TIL and I totally overlooked that part of the documentation. I've revert everything back to using `DynamicUser` and we should now be good to merge.